pkg/support/dump.go

```diff
@@ -104,10 +107,52 @@ func dumpNetworkPolicyResources(fs afero.Fs, executor exec.Interface, basedir st
 	return dumpAntctlGet(fs, executor, "addressgroups", basedir)
 }
 
-// fileCopy copies files under the srcDir to the targetDir. Only files whose
-// name matches the prefixFilter will be copied. Copied files will be located
-// under the same relative path.
-func fileCopy(fs afero.Fs, targetDir string, srcDir string, prefixFilter string) error {
+func timestampFilter(since string) *time.Time {
+	var timeFilter *time.Time
+	if since != "" {
+		duration, _ := time.ParseDuration(since)
+		start := time.Now().Add(-duration)
+		timeFilter = &start
+	}
+	return timeFilter
+
+}
+
+// parseTimeFromFileName parse time from log file name.
+// example log file format: <component>.<hostname>.<user>.log.<level>.<yyyymmdd>-<hhmmss>.1
+func parseTimeFromFileName(name string) (time.Time, error) {
+	ss := strings.Split(name, ".")
+	ts := ss[len(ss)-2]
+	return time.Parse("20060102-150405", ts)
+
+}
+
+// parseTimeFromLogLine parse timestamp from the log line.
+// example(kubelet/agent/controller): "I0817 06:55:10.804384       1 shared_informer.go:270] caches populated"
+// example(ovs): "2021-06-02T16:18:52.285Z|00004|reconnect|INFO|unix:/var/run/openvswitch/db.sock: connecting..."
+// the first char indicates the log level.
+func parseTimeFromLogLine(log string, year string, prefix string) (time.Time, error) {
+	ss := strings.Split(log, ".")
+	if ss[0] == "" {
+		return time.Time{}, fmt.Errorf("log line is empty")
+	}
+
+	dateStr := year + ss[0][1:]
+	layout := "20060102 15:04:05"
```

Comment:
I don't get where we get the layout in this line. It seems that the layout should be `I0817 06:55:10` as the example on line 131?

Reply:
Since we remove the leading log level char(`I`) and add year value to the date str, so it became this.

Reply:
@mengdie-song If your question was about why it's 20060102 15:04:05 but not other time. It's required by `time.Parse`. See https://pkg.go.dev/time#pkg-constants and https://pkg.go.dev/time#Parse

Reply:
Thanks for your answers.